vignettes/SEIRD_Model_Compartments.Rmd

```diff
@@ -0,0 +1,249 @@
+
+## The `SEIaImIsRD` class
+In this section we will walk through an example of how to use the `SEIaImIsRD`
+class.
+
+### Initialization of the model
+To create a new `SEIaImIsRD` object, run:
+```{r}
+model <- SEIaImIsRD()
+```
+
+Next, we set the initial conditions and assign the values to the model. Because the populations are normalized, the populations must sum to one.
+```{r}
+S <- 0.99
+E <- 0.01
+I_asymptomatic <- 0
+I_mild <- 0
+I_severe <- 0
+R <- 0
+D <- 0
+
+initial_conditions(model) <- list(S = S, E = E, I_asymptomatic = I_asymptomatic,
+                                  I_mild = I_mild, I_severe = I_severe, R = R, D = D)
+```
+
+We set the parameter values, and assign them to the model.
+```{r}
+beta <- list(i_asymptomatic = 0.80, i_mild = 0.90, i_severe = 1.00)
+kappa <- 0.50
+omega <- 0.01
+p_symptom <- list(i_mild = 0.35, i_severe = 0.05)
+gamma <- list(i_asymptomatic = 0.9, i_mild = 0.50, i_severe = 0.05)
+mu <- list(i_asymptomatic = 0.005, i_mild = 0.05, i_severe = 0.30)
+
+transmission_parameters(model) <- list(beta = beta, kappa = kappa, omega = omega,
+                                       p_symptom = p_symptom, gamma = gamma, mu = mu)
+```
+
+Define helper function for plotting output dataframe of simulation result
+```{r}
+plot_dataframe <- function(dataframe, x = "time", y = "value", c = "compartment") {
+  p <- ggplot(dataframe, aes_string(x = x, y = y)) +
+    geom_line(aes_string(colour = c)) +
+    theme_classic()
+  return(p)
+}
+```
+
+```{r include=FALSE, echo=FALSE}
+# Define helper function for printing out parameter values
+# print_table_transmission_parameters <- function(model){
+#   df.params <- model@transmission_parameters %>%
+#     data.frame() %>%
+#     t()
+#   colnames(df.params) <- "value"
+#   table.params <- kable(df.params) %>%
+#     kable_styling(bootstrap_options = "striped", full_width = FALSE, position="left")
+#   return (table.params)
+# }
+```
+
+
+Lastly, we set time span and time step for the simulation:
+```{r}
+t <- seq(0, 2000, by = 1)
+```
+Here, it makes sense to go with a time step of one as data for an epidemic, such as number of cases and deaths, is often reported per day, so time resolution of hours or minutes is not useful.
+
+The system of ODEs is solved by calling the function `run(model,t)`. In the next few sections, we run the model under different assumption and plot the results. The function `R0(model)` can be used to compute the basic reproduction number for the model with a given set of parameters. 
+
+#### 1. Including natural loss of immunity
+We first assume that immunity does not last forever, so individuals that have recovered from the disease eventually return to the susceptible population. This is called waning immunity and is included in the model when $\omega>0$, as we assigned it above.
+```{r, fig.height = 5, fig.width = 5, fig.align = "center", fig.show="hold"}
+par(mfrow=c(1,2))
+model <- run(model, t)
+model <- R0(model)
```

Comment:
It's confusing that you rename model to be the new R0? Also, perhaps put the call to R0 in the next block near where it's printed?

Reply:
No, it is not renamed. `R0()` is a class-specific function that returns an SEIaImIsRD object. Then `model@R0` get the actual value of R0. As a formal class definition, I set the all the actual functions for the class to return an object of the class, except for validation functions.